chain/network/src/routing_table_actor.rs

```diff
@@ -0,0 +1,293 @@
+use std::collections::HashMap;
+
+use actix::dev::MessageResponse;
+use actix::{Actor, Handler, Message, SyncContext, System};
+#[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+use tracing::error;
+
+use near_performance_metrics_macros::perf;
+use near_primitives::network::PeerId;
+
+#[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+use crate::ibf::{Ibf, IbfBox};
+#[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+use crate::ibf_peer_set::IbfPeerSet;
+#[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+use crate::ibf_set::IbfSet;
+use crate::routing::Edge;
+#[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+use crate::routing::SimpleEdge;
+#[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+use crate::routing::ValidIBFLevel;
+#[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+use crate::routing::MIN_IBF_LEVEL;
+use crate::types::StopMsg;
+#[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+use crate::types::{PartialSync, RoutingState, RoutingVersion2};
+
+/// Actor that maintains routing table information.
+/// TODO (PIOTR, #4859) Finish moving routing table computation to new thread.
+#[derive(Default)]
+pub struct RoutingTableActor {
+    /// Data structures with all edges.
+    edges: HashMap<(PeerId, PeerId), Edge>,
+    /// Data structure used for exchanging routing tables.
+    #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+    pub peer_ibf_set: IbfPeerSet,
+}
+
+#[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+impl RoutingTableActor {
+    pub fn split_edges_for_peer(
+        &self,
+        peer_id: &PeerId,
+        unknown_edges: &[u64],
+    ) -> (Vec<SimpleEdge>, Vec<u64>) {
+        self.peer_ibf_set.split_edges_for_peer(peer_id, unknown_edges)
+    }
+}
+
+impl Handler<StopMsg> for RoutingTableActor {
+    type Result = ();
+    fn handle(&mut self, _: StopMsg, _ctx: &mut Self::Context) -> Self::Result {
+        System::current().stop();
+    }
+}
+
+impl Actor for RoutingTableActor {
+    type Context = SyncContext<Self>;
+}
+
+#[derive(Debug)]
+pub enum RoutingTableMessages {
+    AddEdges(Vec<Edge>),
+    RemoveEdges(Vec<Edge>),
+    #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+    AddPeerIfMissing(PeerId, Option<u64>),
+    #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+    RemovePeer(PeerId),
+    #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+    ProcessIbfMessage {
+        peer_id: PeerId,
+        ibf_msg: RoutingVersion2,
+    },
```

Comment:
It is better to move feature gated variants to the end of all variants so that it is easier to see which ones are gated

Reply:
Done